src/freenet/node/RequestSender.java

```diff
@@ -1716,9 +1734,18 @@ private boolean finishOpennet(PeerNode next) {
 				// RequestHandler will send a noderef back up, eventually, and will unlockHandler() after that point.
 				// But if this is a local request, we need to send the ack now.
 				// Serious race condition not possible here as we set it.
-				if(source == null)
-					ackOpennet(next);
-				else if(origTag.shouldStop()) {
+				if(source == null) {
+				    long delay = randomDelayFinishOpennetLocal();
+				    if(logMINOR) Logger.minor(this, "Delaying opennet completion for "+TimeUtil.formatTime(delay, 2, true));
+				    node.ticker.queueTimedJob(new Runnable() {
+
+                        @Override
+                        public void run() {
+                            ackOpennet(next);
+                        }
+				        
+				    }, delay);
+				} else if(origTag.shouldStop()) {
```

Comment:
This is the important bit: We call ackOpennet() after a computed delay, not immediately.